### PR TITLE
ISPN-12103 Improve ContainerDatabase to add custom wait strategy

### PR DESCRIPTION
**Pull request by gustavolira**
https://issues.redhat.com/browse/ISPN-12103

### CODE REVIEW

File: server/testdriver/core/src/main/java/org/infinispan/server/test/core/persistence/ContainerDatabase.java [hunk 1]
```diff
@@ -34,7 +37,17 @@
                      .env(env)
                      .build();
             });
-      container = new GenericContainer(image).withExposedPorts(port).waitingFor(Wait.forListeningPort());
+      container = new GenericContainer(image)
+            .withExposedPorts(port)
+            .withPrivilegedMode(true)
+            .waitingFor(Wait.forListeningPort());
+
+      String logMessageWaitStrategy = properties.getProperty("database.container.log.regex");
```
Comment (tristantarrant) [inline server/testdriver/core/src/main/java/org/infinispan/server/test/core/persistence/ContainerDatabase.java:45]: The property should be documented and added to the `TestSystemPropertyNames` class

Comment (gustavolira) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/core/persistence/ContainerDatabase.java]: @tristantarrant updated

Comment (gustavolira) [reply server/testdriver/core/src/main/java/org/infinispan/server/test/core/persistence/ContainerDatabase.java]: the PR is dirty, I will fix it

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**